packages/ember-glimmer/tests/integration/syntax/each-test.js

```diff
@@ -171,6 +394,28 @@ moduleFor('Syntax test: {{#each as}}', class extends EachTest {
     this.assertText('Stef-Yehuda-Stef');
   }
 
+  ['@htmlbars {{#each}} can access properties of \'this\' to iterate over']() {
+    this.render(`{{#each this.list as |item|}}{{item.name}}{{/each}}`, {
+      list: emberA([{ name: 'Chad' }, { name: 'Zack' }, { name: 'Asa' }])
+    });
+
+    this.assertText('ChadZackAsa');
+
+    this.assertStableRerender();
+
+    this.runTask(() => get(this.context, 'list').replace(1, 1, { name: 'Joel' }));
+
+    this.assertText('ChadJoelAsa');
+
+    this.runTask(() => set(this.context, 'list', [{ name: 'Yehuda' }, { name: 'Godfrey' }, { name: 'Matt' }]));
```

Comment:
I wouldn't use emberA for the 'R' (replace) part, it is nice to test that you can use a plain array with each.  I would use emberA() just for the 'U' (update) part, since it is required there for the interior mutations.